### PR TITLE
FIX: [droid] Samba 3.6.12 (gplv3)

### PR DESCRIPTION
**Pull request by koying**
/cc @wsnipex

### CODE REVIEW

**Comment by koying** [comment]
jenkins build this please

File: tools/android/packaging/xbmc/src/org/xbmc/kodi/Splash.java.in [hunk 1]
```diff
@@ -544,8 +544,13 @@ public class Splash extends Activity {
...
   protected void startXBMC() {
-    // NB: We only preload lib@APP_NAME_LC@ to be able to get info on missing symbols.
-    //     This is not normally needed
+    // The dynamic linker on droid is a mess.
+    // Depending on version, it might not be able to properly resolve
+    try
+    {
+      System.loadLibrary("smbclient");
+    }
+    catch (UnsatisfiedLinkError e) {}
```
Comment (wsnipex) [inline tools/android/packaging/xbmc/src/org/xbmc/kodi/Splash.java.in]: would a log msg make sense here?

**Comment by wsnipex** [comment]
please adjust https://github.com/xbmc/xbmc/pull/6833/files#diff-25bb87ae2148a0a2abba0a8a4296b5bfL54

**Comment by wsnipex** [comment]
looks good, the LFS support looks neat :)

File: tools/android/packaging/xbmc/src/org/xbmc/kodi/Splash.java.in (hunk 1 above)
Comment (koying) [inline tools/android/packaging/xbmc/src/org/xbmc/kodi/Splash.java.in]: Not really. 3.0 android was static, while everyone uses 3.6 dynamic apparently. The try-catch allows to still run with the static lib.
I shouldn't have to preload the lib, really, but the droid dloader is not behaving on, e.g. aftv.

Comment (wsnipex) [inline tools/android/packaging/xbmc/src/org/xbmc/kodi/Splash.java.in]: ah, understood, thanks. It wasn't immediately obvious to me that this is there to support shared and static libs. Maybe add this to the comment?

**Comment by koying** [comment]
jenkins build this please

**Comment by MartijnKaijser** [comment]
users could still compile with the gplv2 version but we compile by default with gplv3?
time to get this in?

**Comment by koying** [comment]
Not many actually cared to use the test build, so I assume yes, and let's see what comes out ;)
I'll merge in 24h if no -1 (I still have to squash)

**Comment by MartijnKaijser** [comment]
i tested it and had no issues ;)

**Comment by koying** [comment]
jenkins build and merge

**Pull request merged**